extensions/mailer/runtime/src/main/java/io/quarkus/mailer/impl/MailConfigRecorder.java

```diff
@@ -41,10 +42,13 @@
     }
 
     void initialize(Vertx vertx, MailConfig config) {
-        if (client != null) {
+        if (client != null && config.equals(MailConfigRecorder.config)) {
```

Comment:
@cescoffier @machi1990 I'm not totally convinced by this change.

Shouldn't we properly close the client on shutdown and start a new one each time?

Reply:
> Shouldn't we properly close the client on shutdown and start a new one each time?

This was my other option but creating a client on each reload even though unnecessary was not right to me.

Reply:
Well, we usually don't do fancy things like that in other extensions and TBH I'm not sure I like it.

The contract we have for dev mode was: let's restart your Quarkus fully. And if we begin having all sorts of edge cases like that, I'm not sure it won't be a Pandora box of future bugs.

Reply:
But I'll let @cescoffier decide, I just wanted to raise my concerns.

Reply:
> The contract we have for dev mode was: let's restart your Quarkus fully. 

Sounds reasonable enough. @cescoffier WDYT?